### ISSUE TITLE
certificated relation hook error

### ISSUE THREAD

**Issue by javacruft** [juju-solutions/layer-etcd#72]
While deploying the canonical-k8s spell using conjure-up, I saw the following error from one of the etcd units:

```
017-01-13 07:56:44 INFO juju-log certificates:1: Reactive main running for hook certificates-relation-changed
2017-01-13 07:56:45 INFO juju-log certificates:1: Initializing Leadership Layer (is follower)
2017-01-13 07:56:45 INFO juju-log certificates:1: Invoking reactive handler: hooks/relations/tls-certificates/requires.py:22:changed
2017-01-13 07:56:46 INFO juju-log certificates:1: Invoking reactive handler: reactive/etcd.py:67:set_app_version
2017-01-13 07:56:47 INFO juju-log certificates:1: Invoking reactive handler: reactive/tls_client.py:32:store_server
2017-01-13 07:56:47 INFO juju-log certificates:1: Writing server certificate to /etc/ssl/etcd/server.crt
2017-01-13 07:56:47 INFO juju-log certificates:1: Writing server key to /etc/ssl/etcd/server.key
2017-01-13 07:56:47 INFO juju-log certificates:1: Invoking reactive handler: reactive/tls_client.py:14:store_ca
2017-01-13 07:56:47 INFO juju-log certificates:1: Invoking reactive handler: reactive/tls_client.py:53:store_client
2017-01-13 07:56:48 INFO juju-log certificates:1: Invoking reactive handler: reactive/etcd.py:85:prepare_tls_certificates
2017-01-13 07:56:50 INFO juju-log certificates:1: Invoking reactive handler: reactive/etcd.py:98:spam_ownership_of_tls_certs
2017-01-13 07:56:51 INFO juju-log certificates:1: Invoking reactive handler: reactive/etcd.py:272:register_node_with_leader
2017-01-13 07:56:51 INFO certificates-relation-changed Error:  dial tcp 127.0.0.1:2379: getsockopt: connection refused
2017-01-13 07:56:52 INFO certificates-relation-changed Traceback (most recent call last):
2017-01-13 07:56:52 INFO certificates-relation-changed   File "/var/lib/juju/agents/unit-etcd-1/charm/hooks/certificates-relation-changed", line 19, in <module>
2017-01-13 07:56:52 INFO certificates-relation-changed     main()
2017-01-13 07:56:52 INFO certificates-relation-changed   File "/usr/local/lib/python3.5/dist-packages/charms/reactive/__init__.py", line 78, in main
2017-01-13 07:56:52 INFO certificates-relation-changed     bus.dispatch()
2017-01-13 07:56:52 INFO certificates-relation-changed   File "/usr/local/lib/python3.5/dist-packages/charms/reactive/bus.py", line 434, in dispatch
2017-01-13 07:56:52 INFO certificates-relation-changed     _invoke(other_handlers)
2017-01-13 07:56:52 INFO certificates-relation-changed   File "/usr/local/lib/python3.5/dist-packages/charms/reactive/bus.py", line 417, in _invoke
2017-01-13 07:56:52 INFO certificates-relation-changed     handler.invoke()
2017-01-13 07:56:52 INFO certificates-relation-changed   File "/usr/local/lib/python3.5/dist-packages/charms/reactive/bus.py", line 291, in invoke
2017-01-13 07:56:52 INFO certificates-relation-changed     self._action(*args)
2017-01-13 07:56:52 INFO certificates-relation-changed   File "/var/lib/juju/agents/unit-etcd-1/charm/reactive/etcd.py", line 297, in register_node_with_leader
2017-01-13 07:56:52 INFO certificates-relation-changed     peers = etcdctl.member_list(leader_get('leader_address'))
2017-01-13 07:56:52 INFO certificates-relation-changed   File "lib/etcdctl.py", line 75, in member_list
2017-01-13 07:56:52 INFO certificates-relation-changed     out = self.run("etcdctl member list")
2017-01-13 07:56:52 INFO certificates-relation-changed   File "lib/etcdctl.py", line 139, in run
2017-01-13 07:56:52 INFO certificates-relation-changed     return check_output(split(command)).decode('ascii')
2017-01-13 07:56:52 INFO certificates-relation-changed   File "/usr/lib/python3.5/subprocess.py", line 626, in check_output
2017-01-13 07:56:52 INFO certificates-relation-changed     **kwargs).stdout
2017-01-13 07:56:52 INFO certificates-relation-changed   File "/usr/lib/python3.5/subprocess.py", line 708, in run
2017-01-13 07:56:52 INFO certificates-relation-changed     output=stdout, stderr=stderr)
2017-01-13 07:56:52 INFO certificates-relation-changed subprocess.CalledProcessError: Command '['etcdctl', 'member', 'list']' returned non-zero exit status 4
2017-01-13 07:56:52 ERROR juju.worker.uniter.operation runhook.go:107 hook "certificates-relation-changed" failed: exit status 1
```

Juju automatically retried the hook twice, at which point it succeeded, but the unit reports:

```
etcd/0                    active    idle   1        10.30.4.169     2379/tcp        Healthy with 3 known peers.
etcd/1                    active    idle   2        10.30.4.124     2379/tcp        Healthy with 0 known peers.
etcd/2*                   active    idle   3        10.30.4.197     2379/tcp        Healthy with 3 known peers.
```

Deployment was using the Juju LXD provider on the a local machine (Ubuntu zesty development).

**Comment by axinojolais** [comment]
This is a duplicate of https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/153

**Comment by chuckbutler** [comment]
This was fixed in #153, it is scheduled to be released today. Sorry you encountered this issue, and thank you for filing the bug
